bridge/setu/broadcaster/broadcaster.go

```diff
@@ -52,41 +59,35 @@ func (tb *TxBroadcaster) BroadcastToHeimdall(msg sdk.Msg) error {
 	txEncoder := helper.GetTxEncoder(tb.cliCtx.Codec)
 	// chain id
 	chainID := helper.GetGenesisDoc().ChainID
-	// current address
-	address := hmTypes.BytesToHeimdallAddress(helper.GetAddress())
+
 	// fetch from APIs
-	var account authTypes.Account
-	response, err := util.FetchFromAPI(tb.cliCtx, util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, address)))
+	account, err := util.GetAccount(tb.cliCtx)
 	if err != nil {
-		tb.logger.Error("Error fetching account from rest-api", "url", util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, address)))
-		panic("Error connecting to rest-server, please start server before bridge.")
-	}
-
-	// get proposer from response
-	if err := tb.cliCtx.Codec.UnmarshalJSON(response.Result, &account); err != nil && len(response.Result) != 0 {
-		tb.logger.Error("Error unmarshalling account details", "url", util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, address)))
+		tb.logger.Error("Error fetching account from rest-api", "url", util.GetHeimdallServerEndpoint(fmt.Sprintf(util.AccountDetailsURL, helper.GetAddress())))
 		return err
 	}
-
 	// get account number and sequence
 	accNum := account.GetAccountNumber()
-	accSeq := account.GetSequence()
 	txBldr := authTypes.NewTxBuilderFromCLI().
 		WithTxEncoder(txEncoder).
 		WithAccountNumber(accNum).
-		WithSequence(accSeq).
+		WithSequence(tb.lastSeqNo).
 		WithChainID(chainID)
 
 	txResponse, err := helper.BuildAndBroadcastMsgs(tb.cliCtx, txBldr, []sdk.Msg{msg})
 	if err != nil {
 		tb.logger.Error("Error while broadcasting the heimdall transaction", "error", err)
+
+		// update seqNo for safety
+		tb.lastSeqNo = account.GetSequence()
+
 		return err
 	}
 
-	// increment account sequence
-	// accSeq = accSeq + 1
-	tb.logger.Info("Tx sent on heimdall", "txHash", txResponse.TxHash, "accSeq", accSeq, "accNum", accNum)
+	tb.logger.Info("Tx sent on heimdall", "txHash", txResponse.TxHash, "accSeq", tb.lastSeqNo, "accNum", accNum)
 	tb.logger.Debug("Tx successful on heimdall", "txResponse", txResponse)
+	// increment account sequence
+	tb.lastSeqNo += 1
```

Comment:
If any error, use sequence directly. So this statement shouldn't be in else statement.

Reply:
If there is any error, i am setting the lastSeqNo from account which is fetched from rest-server. 
returning back the error.  when error is returned, that transaction will be retried later.

line no 90 won't be executed when there is an error.

Reply:
Yeah great.